examples/chef/common/chef-fan-control-manager.cpp

```diff
@@ -99,98 +124,210 @@ Status ChefFanControlManager::HandleStep(StepDirectionEnum aDirection, bool aWra
     return SpeedSetting::Set(mEndpoint, newSpeedSetting);
 }
 
-CHIP_ERROR ChefFanControlManager::Write(const ConcreteDataAttributePath & aPath, AttributeValueDecoder & aDecoder)
+void ChefFanControlManager::HandleFanControlAttributeChange(AttributeId attributeId, uint8_t type, uint16_t size, uint8_t * value)
 {
-    VerifyOrDie(aPath.mClusterId == FanControl::Id);
-    VerifyOrDie(aPath.mEndpointId == mEndpoint);
+    ChipLogProgress(NotSpecified, "ChefFanControlManager::HandleFanControlAttributeChange");
+    switch (attributeId)
+    {
+    case FanControl::Attributes::PercentSetting::Id: {
+        ChipLogProgress(NotSpecified, "ChefFanControlManager::HandleFanControlAttributeChange  PercentSetting");
+        DataModel::Nullable<Percent> percentSetting =
+            NumericAttributeTraits<DataModel::Nullable<Percent>>::StorageToWorking(*value);
```

Comment:
The fact that this compiles at all is sort of broken..... It does not do the right thing, for sure:
```
NumericAttributeTraits<DataModel::Nullable<Percent>>::StorageToWorking(255).IsNull()
```
returns false, right?

I think you want something more like this:
```suggestion
        DataModel::Nullable<Percent> percentSetting;
        if (!NumericAttributeTraits<Percent>::IsNullValue(*value)) {
            percentSetting.SetNonNull(NumericAttributeTraits<Percent>::StorageToWorking(*value));
        }
```

Reply:
https://github.com/project-chip/connectedhomeip/pull/36459 to make this not compile.

Reply:
Good catch, Thanks! Yes, I assumed that would convert properly. Thanks for the follow up PR to make this mistake no longer possible.